console/console-init/ui/src/modules/iot-device/components/DeviceFilter/DeviceFilter.tsx

```diff
@@ -220,6 +288,80 @@ const DeviceFilter: React.FunctionComponent<IDeviceFilterProps> = ({
         <FormGroup label="Added date" fieldId="filter-device-added-date">
           <DateFilterSection filter={filter} setFilter={setFilter} />
         </FormGroup>
+        <FormGroup
+          label="Gateway group membership"
+          fieldId="device-filter-gateway-membership-group-input"
+        >
+          <GatewayGroupTypeAheadSelect
+            id="add-gateway-group-membership-typeahead-select"
+            aria-label="gateway group membership dropdown"
+            aria-describedby="multi typeahead for gateway groups membership"
+            onSelect={onSelectGatewayGroup}
+            onClear={onClear}
+            selected={gatewayGroups}
+            typeAheadAriaLabel={"typeahead to select gateway group membership"}
+            isMultiple={true}
+            placeholderText={"Input gateway group name"}
+          />
+        </FormGroup>
+
+        <FormGroup
+          label="Gateway connections"
+          fieldId="device-gateway-connection-dropdowntoggle"
+        >
+          <Grid>
+            <GridItem span={4}>
+              <DropdownWithToggle
+                id={"device-filter-gateway-conn-dropdown"}
+                toggleId="device-filter-gateway-conn-dropdowntoggle"
+                name="Device Connections"
+                className={css(styles.dropdown_align)}
+                toggleClass={css(styles.dropdown_toggle_align)}
+                position={DropdownPosition.left}
+                onSelectItem={onGatewayConnectionSelect}
+                dropdownItems={deviceGatewayConnectionOptions}
+                value={gatewayGroupConnection}
+                isLabelAndValueNotSame={true}
+              />
+            </GridItem>
+            <GridItem span={5} className={css(styles.grid_margin)}>
+              <TextInput
+                isRequired
+                type="text"
+                id="device-filter-gateway-connection-input"
+                placeholder="Input gateway"
+                name="gateway-connection"
+                aria-describedby="gateway connection for filter"
+                value={gatewayInput}
+                onChange={onChangeGatewayConnection}
+              />
+            </GridItem>
+            <GridItem span={3} className={css(styles.grid_margin)}>
+              <Button
+                variant={ButtonVariant.primary}
+                id="device-filter-add-gateway-conn-btn"
+                isDisabled={
+                  gatewayGroupConnection.trim() === "" ||
+                  gatewayInput.trim() === ""
+                }
+                onClick={addGatewayConnection}
+              >
+                Add
+              </Button>
+            </GridItem>
+          </Grid>
+          <br />
+          <ChipGroup>
+            {gatewayConnections.map(currentChip => (
```

Comment:
currentChip is not a good name, please change it. 
You should replace this chip group logic with custom component ChipGroupsWithTitle which will available after merging PR#5145